src/main/java/com/exactpro/th2/crawler/state/v1/RecoveryStateProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.crawler.state.v1;
+
+import com.exactpro.th2.crawler.state.BaseState;
+import com.exactpro.th2.crawler.state.StateProvider;
+import com.exactpro.th2.crawler.state.StdStateConverter;
+import com.exactpro.th2.crawler.state.Version;
+import com.google.auto.service.AutoService;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@AutoService(StateProvider.class)
+public class RecoveryStateProvider implements StateProvider {
+    @NotNull
+    @Override
+    public Version getVersion() {
+        return Version.V_1;
+    }
+
+    @NotNull
+    @Override
+    public Class<RecoveryState> getStateClass() {
+        return RecoveryState.class;
+    }
+
+    @Nullable
+    @Override
+    public StdStateConverter<RecoveryState, BaseState> getConverter() {
+        return null;
```

Comment:
Why do we need to write a warning into the log? I would prefer to have an error in this case because if we don't have a converter we cannot correctly start the crawler from the current state. And we should not proceed to the next interval.

Also, I want to say that this state provider does not have a converter. The `null` is the best option here in my opinion

Reply:
Maybe write default implementation which throws an exception. We can use it for the last version only

Reply:
I still don't see any reason to do so. The state service will throw an exception if the version is not the last one and the provider for that version does not have a converter

Reply:
Add the constraint: method should return a not null value, but this approach can be more complicated in written logic related to check not null.